markless.el

```diff
@@ -355,7 +355,30 @@ Marks PRE and POST as markup and the content with PROP."
                              nil nil nil nil
                              (font-lock-multiline . t)))
   (setq-local flyspell-generic-check-word-predicate
-              #'markless-at-word-p))
+              #'markless-at-word-p)
+  (setq-local comment-start ";")
+  (setq-local paragraph-start
+              (mapconcat #'identity
+                         '(
+                           "\f" ; linefeed
+                           "[ \t\f]*$" ; whitespace-only line
+                           "[ \t]*\|[ \t\f]*$" ; empty blockquote line
+                           "[ \t]*-[ \t]+" ; unordered list item
+                           "[ \t]*[0-9]+\\.[ \t]+" ; ordered list item
+                           "\\[[0-9]+\\]" ; Footnote
+                           )
+                         "\\|"))
+  (setq-local paragraph-separate
+              (mapconcat #'identity
+                         '(
+                           "[ \t\f]*$"  ; whitespace-only line
+                           "[ \t]*\|[ \t\f]*$" ; empty blockquote line
+                           "[ \t]*~[ \t\f]*" ; blockquote header
+                           "^#+" ; heading
+                           "^! " ; instruction
+                           "^; " ; comment
```

Comment:
You forgot about:
- embeds `"[ "`
- horizontal rules `"==+"`
- footnotes `"[\d+] "`

The comment should also be `";+ "`, and remember that you can have a heading, instruction, etc. within another directive like a blockquote, so I'm not sure that the `^` there is appropriate.

Reply:
I've adjusted the regular expressions to also work in blockquotes and indented paragraphs, and added embeds, horizontal rules and footnotes.